app/models/services/base-service.ts

```diff
@@ -403,6 +403,23 @@ export default abstract class BaseService<ServiceParamType> {
     return this.numInputGranules;
   }
 
+  /**
+   * Return the number of actual workflow steps for this request
+   *
+   * @returns  the number of actual workflow steps
+   */
+  protected _numActualSteps(): number {
+    let totalSteps = 0;
+    if (this.config.steps) {
+      this.config.steps.forEach(((step) => {
```

Comment:
You're forgetting your functional roots, Leo, This is the perfect use case for `reduce`.

Reply:
I am just not familiar with `reduce` in javascript. As a result, the `foreach` is a lot more readable for me. :)